litestar/response/redirect.py

```diff
@@ -1,71 +1,139 @@
 from __future__ import annotations
 
 from typing import TYPE_CHECKING, Any, Literal
-from urllib.parse import quote
 
 from litestar.constants import REDIRECT_ALLOWED_MEDIA_TYPES, REDIRECT_STATUS_CODES
 from litestar.enums import MediaType
 from litestar.exceptions import ImproperlyConfiguredException
-from litestar.response.base import Response
+from litestar.response.base import ASGIResponse, Response
 from litestar.status_codes import HTTP_307_TEMPORARY_REDIRECT
-
-__all__ = ("RedirectResponse",)
-
+from litestar.utils import url_quote
+from litestar.utils.helpers import filter_cookies, get_enum_string_value
 
 if TYPE_CHECKING:
+    from litestar.app import Litestar
     from litestar.background_tasks import BackgroundTask, BackgroundTasks
-    from litestar.types import ResponseCookies
+    from litestar.connection import Request
+    from litestar.datastructures import Cookie
+    from litestar.types import ResponseCookies, ResponseHeaders, TypeEncodersMap
+
+__all__ = (
+    "ASGIRedirectResponse",
+    "RedirectResponse",
+)
+
+
+RedirectStatusType = Literal[301, 302, 303, 307, 308]
+"""Acceptable status codes for redirect responses."""
+
+
+class ASGIRedirectResponse(ASGIResponse):
+    """A low-level ASGI redirect response class."""
+
+    def __init__(
+        self,
+        url: str | bytes,
```

Comment:
I think `path` is correct, since it does not have to be an actual URL.

Reply:
Agree - this was `path` on the container type and `url` on the response type. I'll change.